src/clib/initialize_chemistry_data.c

```diff
@@ -151,6 +248,9 @@ int local_initialize_chemistry_data(chemistry_data *my_chemistry,
   if (my_chemistry->h2_on_dust > 0 || my_chemistry->dust_chemistry > 0 || my_chemistry->dust_recombination_cooling > 0) {
     my_rates->gas_grain = malloc(my_chemistry->NumberOfTemperatureBins * sizeof(double));
     my_rates->regr      = malloc(my_chemistry->NumberOfTemperatureBins * sizeof(double));
+  } else {
+    my_rates->gas_grain = NULL;
+    my_rates->regr = NULL;
   }
```

Comment:
I would appreciate someone double-checking this, but I actually think this whole block can be removed. `regr` and `gas_grain` are being set to `NULL` in the changes above, but these rates are being set with the `add_reaction_rate` function, which also does the malloc. The code as it stands has a memory leak.